src/main/kotlin/no/nav/familie/ba/sak/behandling/resultat/YtelsePersonUtils.kt

```diff
@@ -0,0 +1,148 @@
+package no.nav.familie.ba.sak.behandling.resultat
+
+import no.nav.familie.ba.sak.behandling.restDomene.SøknadDTO
+import no.nav.familie.ba.sak.beregning.domene.AndelTilkjentYtelse
+import no.nav.familie.ba.sak.beregning.domene.YtelseType
+import no.nav.familie.ba.sak.beregning.domene.erLøpende
+import no.nav.familie.ba.sak.common.Feil
+import no.nav.familie.ba.sak.common.TIDENES_MORGEN
+import no.nav.familie.ba.sak.common.førsteDagIInneværendeMåned
+import no.nav.familie.ba.sak.common.sisteDagIInneværendeMåned
+import no.nav.familie.ba.sak.common.toYearMonth
+import no.nav.fpsak.tidsserie.LocalDateSegment
+import no.nav.fpsak.tidsserie.LocalDateTimeline
+import java.time.YearMonth
+
+object YtelsePersonUtils {
+
+    /**
+     * Metode for å utlede kravene for å utlede behandlingsresultat per krav.
+     * Metoden finner kravene som ble stilt i søknaden,
+     * samt ytelsestypene per person fra forrige behandling.
+     */
+    fun utledKrav(søknadDTO: SøknadDTO?,
+                  forrigeAndelerTilkjentYtelse: List<AndelTilkjentYtelse>): List<YtelsePerson> {
+        val ytelsePersoner: MutableSet<YtelsePerson> =
+                søknadDTO?.barnaMedOpplysninger?.filter { it.inkludertISøknaden }?.map {
+                    YtelsePerson(personIdent = it.ident,
+                                 ytelseType = YtelseType.ORDINÆR_BARNETRYGD,
+                                 erFramstiltKravForINåværendeBehandling = true)
+                }?.toMutableSet() ?: mutableSetOf()
+
+        forrigeAndelerTilkjentYtelse.forEach {
+            val nyYtelsePerson = YtelsePerson(
+                    personIdent = it.personIdent,
+                    ytelseType = it.type,
+                    erFramstiltKravForINåværendeBehandling = false
+            )
+
+            if (!ytelsePersoner.contains(nyYtelsePerson)) {
+                ytelsePersoner.add(nyYtelsePerson)
+            }
+        }
+
+        return ytelsePersoner.toList()
+    }
```

Comment:
nyYtelsePerson er vel egentlig gammelYtelsePerson som skal legges til med et false-flagg. Det er heller ikke så tydelig her at contains ser på hash av kun ident og type.

Er det enklere å forstå som noe sånt? 

```suggestion
val personerFramstiltKravForITidligere: List<YtelsePerson> = forrigeAndelerTilkjentYtelse.map{ it.tilYtelsePerson(it.personident, it.type)}
```

Hvor man på AndelTilkjentYtelse lager extension
```
fun tilYtelsePerson(erFramstiltKravForINåværendeBehandling: Boolean = false) =
            YtelsePerson(personIdent = this.personIdent,
                         ytelseType = this.type,
                         erFramstiltKravForINåværendeBehandling = erFramstiltKravForINåværendeBehandling)
```

og til slutt her returner noe lignende
`return listOf(personerFramstiltKravForINå, personerFramstiltKravForTidligere.filter(!personerFramstiltKravForINå.contains(it)).flatten()`

Reply:
ikke sikkert det er så mye mer lesbart kanskje :P

Reply:
Nå er dette ren flytting, men jeg kan ta en titt 💪